src/web/metrics.rs

```diff
@@ -2,119 +2,149 @@ use crate::db::Pool;
+pub static UPLOADED_FILES_TOTAL: Lazy<IntCounter> = Lazy::new(|| {
+    register_int_counter!(
         "docsrs_uploaded_files_total",
         "Number of files uploaded to S3 or stored in the database"
     )
-    .unwrap();
+    .unwrap()
+});
 
-    pub static ref ROUTES_VISITED: IntCounterVec = register_int_counter_vec!(
+pub static ROUTES_VISITED: Lazy<IntCounterVec> = Lazy::new(|| {
+    register_int_counter_vec!(
         "docsrs_routes_visited",
         "The traffic of various docs.rs routes",
         &["route"]
     )
-    .unwrap();
+    .unwrap()
+});
 
-    pub static ref RESPONSE_TIMES: HistogramVec = register_histogram_vec!(
+pub static RESPONSE_TIMES: Lazy<HistogramVec> = Lazy::new(|| {
+    register_histogram_vec!(
         "docsrs_response_time",
         "The response times of various docs.rs routes",
         &["route"]
     )
-    .unwrap();
+    .unwrap()
+});
 
-    pub static ref RUSTDOC_RENDERING_TIMES: HistogramVec = register_histogram_vec!(
+pub static RUSTDOC_RENDERING_TIMES: Lazy<HistogramVec> = Lazy::new(|| {
+    register_histogram_vec!(
         "docsrs_rustdoc_rendering_time",
         "The time it takes to render a rustdoc page",
         &["step"]
     )
-    .unwrap();
+    .unwrap()
+});
 
-    pub static ref FAILED_DB_CONNECTIONS: IntCounter = register_int_counter!(
+pub static FAILED_DB_CONNECTIONS: Lazy<IntCounter> = Lazy::new(|| {
+    register_int_counter!(
         "docsrs_failed_db_connections",
         "Number of attempted and failed connections to the database"
     )
-    .unwrap();
+    .unwrap()
+});
 
-    pub static ref USED_DB_CONNECTIONS: IntGauge = register_int_gauge!(
+pub static USED_DB_CONNECTIONS: Lazy<IntGauge> = Lazy::new(|| {
+    register_int_gauge!(
         "docsrs_used_db_connections",
         "The number of used database connections"
     )
-    .unwrap();
+    .unwrap()
+});
 
-    pub static ref IDLE_DB_CONNECTIONS: IntGauge = register_int_gauge!(
+pub static IDLE_DB_CONNECTIONS: Lazy<IntGauge> = Lazy::new(|| {
+    register_int_gauge!(
         "docsrs_idle_db_connections",
         "The number of idle database connections"
     )
-    .unwrap();
+    .unwrap()
+});
 
-    pub static ref MAX_DB_CONNECTIONS: IntGauge = register_int_gauge!(
+pub static MAX_DB_CONNECTIONS: Lazy<IntGauge> = Lazy::new(|| {
+    register_int_gauge!(
         "docsrs_max_db_connections",
         "The maximum database connections"
     )
-    .unwrap();
+    .unwrap()
+});
 
-    pub static ref OPEN_FILE_DESCRIPTORS: IntGauge = register_int_gauge!(
+#[cfg(not(windows))]
+pub static OPEN_FILE_DESCRIPTORS: Lazy<IntGauge> = Lazy::new(|| {
+    register_int_gauge!(
         "docsrs_open_file_descriptors",
         "The number of currently opened file descriptors"
     )
-    .unwrap();
+    .unwrap()
+});
 
-    pub static ref CURRENTLY_RUNNING_THREADS: IntGauge = register_int_gauge!(
+#[cfg(not(windows))]
+pub static CURRENTLY_RUNNING_THREADS: Lazy<IntGauge> = Lazy::new(|| {
+    register_int_gauge!(
```

Comment:
Why was this cfg added?

Reply:
Whoops, thought I answered this yesterday! The cfg was added because after the generated code from `lazy_static` was removed it was revealed that on windows those two metrics aren't used (and therefore threw warnings), which makes sense since the driving code uses `libc`. If you want I can add (in a separate PR) the # of threads for windows, but since file descriptors don't exist on windows I can't add that metric.

Reply:
Hmm I see. Since we don't run windows in prod I'm ok with not having this metric.